File: benchmarks/reference/fib.go

package main

import (
	"fmt"
	"time"
)

func fib(n int) int {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	start := time.Now()
	
	fmt.Printf("Fibonacci(30): %d\n", fib(30))
	
	elapsed := time.Since(start).Seconds()
	fmt.Printf("Time: %.6f\n", elapsed)
}

File: benchmarks/reference/fib.py

import time

def fib(n):
    if n < 2: return n
    return fib(n - 1) + fib(n - 2)

if __name__ == "__main__":
    start = time.time()
    print(f"Fibonacci(30): {fib(30)}")
    elapsed = time.time() - start
    print(f"Time: {elapsed:.6f}")
